run.sh

```diff
@@ -24,6 +32,44 @@ API_TOKEN_PATH=${API_TOKEN_PATH:-/var/run/secrets/kubernetes.io/serviceaccount/t
 KUBE_TOKEN=$(< ${API_TOKEN_PATH})
 KUBERNETES_PORT_443_TCP_ADDR=${KUBERNETES_PORT_443_TCP_ADDR:-kubernetes}
 
+kubectl -n kube-system get secret cilium-etcd-client-tls || {
+	cd certs
+	echo "generating CA certs ==="
+	cfssl gencert -initca ca-csr.json | cfssljson -bare ca
+
+	echo "generating etcd peer.json for cluster domain ${CLUSTER_DOMAIN} ==="
+
+	sed -e "s/CLUSTER_DOMAIN/${CLUSTER_DOMAIN}/" peer.json.sed > peer.json
+
+	echo "generating etcd peer certs ==="
+	cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=peer peer.json | cfssljson -bare peer
+
+	echo "generating etcd server certs ==="
+	cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=server server.json | cfssljson -bare server
+
+	echo "generating etcd client certs ==="
+	cfssl gencert -ca=ca.pem -ca-key=ca-key.pem -config=ca-config.json -profile=client etcd-client.json | cfssljson -bare etcd-client
+
+	mv etcd-client.pem etcd-client.crt
+	mv etcd-client-key.pem etcd-client.key
+	cp ca.pem etcd-client-ca.crt
+
+	mv server.pem server.crt
+	mv server-key.pem server.key
+	cp ca.pem server-ca.crt
+
+	mv peer.pem peer.crt
+	mv peer-key.pem peer.key
+	mv ca.pem peer-ca.crt
+
+	rm *.csr ca-key.pem
+
+	kubectl create secret generic -n kube-system cilium-etcd-peer-tls --from-file=peer-ca.crt --from-file=peer.crt --from-file=peer.key
+	kubectl create secret generic -n kube-system cilium-etcd-server-tls --from-file=server-ca.crt --from-file=server.crt --from-file=server.key
+	kubectl create secret generic -n kube-system cilium-etcd-client-tls --from-file=etcd-client-ca.crt --from-file=etcd-client.crt --from-file=etcd-client.key
```

Comment:
The certificates were uploaded to kubernetes, I think it's reasonable paranoid to delete them from this container.